src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -127,42 +131,132 @@ public void shootAtComputer(int row, int col) {
     }
 
     public void shootAtPlayer() {
-        double randomRow = Math.random() * 10 + 1;
-        double randomCol = Math.random() * 10 + 1;
         int max = 10;
         int min = 1;
         Random random = new Random();
         int randRow = random.nextInt(max - min + 1) + min;
         int randCol = random.nextInt(max - min + 1) + min;
 
         Coordinate coor = new Coordinate(randRow,randCol);
-        if(playerMisses.contains(coor)){
-            System.out.println("Dupe");
+        if(checkAIhit(randRow, randCol) || checkAImiss(randRow, randCol)){
```

Comment:
will have to rename randRow and randCol to rowShoot and colShoot in order to work with my pattern firing